rust/src/datatypes/field.rs

```diff
@@ -546,30 +548,33 @@ impl TryFrom<ArrowField> for Field {
 impl From<&Field> for ArrowField {
     fn from(field: &Field) -> Self {
         let out = Self::new(&field.name, field.data_type(), field.nullable);
-
-        if !field.extension_name.is_empty() {
-            out.with_metadata(
-                vec![(
-                    "ARROW:extension:name".to_string(),
-                    field.extension_name.clone(),
-                )]
-                .into_iter()
-                .collect(),
-            )
-        } else {
-            out
-        }
+        out.with_metadata(field.metadata.clone())
     }
 }
 
 impl From<&pb::Field> for Field {
     fn from(field: &pb::Field) -> Self {
+        let mut lance_metadata: HashMap<String, String> = field
+            .metadata
+            .iter()
+            .map(|(key, value)| {
+                let string_value = String::from_utf8_lossy(value).to_string();
+                (key.clone(), string_value)
+            })
+            .collect();
+        if !field.extension_name.is_empty() {
+            lance_metadata.insert(
+                "ARROW:extension:name".to_string(),
+                field.extension_name.clone(),
+            );
+        }
+        // TODO (check extension_name)
```

Comment:
What's this TODO? can it be removed?

Reply:
Ah, sorry, forgot to remove this (it's resolved by the if check above it).  I have now removed this comment.